frontend/app/controllers/Event.scala

```diff
@@ -56,6 +61,46 @@ trait Event extends Controller with ActivityTracking {
     eventOpt.getOrElse(Redirect(Links.membershipFront))
   }
 
+  case class EmbedData(title: String,
+                       image: Option[String],
+                       venue: Option[String],
+                       location: Option[String],
+                       price: Option[String],
+                       identifier: String,
+                       start: String,
+                       end: String)
+  case class EmbedResponse(status: String, result: Option[EmbedData])
+
+  implicit val writesEmbedData = Json.writes[EmbedData]
+  implicit val writesResponse = Json.writes[EmbedResponse]
+
+  /*
+   * This endpoint is hit by Composer when embedding Membership events.
+   * Changes here will need to be reflected in the flexible-content repo.
+   * Note that Composer will index this data, which is in turn indexed by CAPI.
+   * (eg. updates to event details will not be reflected post-embed)
+   */
+  def embedData(slug: String, callback: String) = CachedAction { implicit request =>
+  val standardFormat = ISODateTimeFormat.dateTime.withZoneUTC
+
+    val eventDataOpt = for {
+      id <- EBEvent.slugToId(slug)
+      event <- EventbriteService.getEvent(id)
+    } yield EmbedData(
```

Comment:
This is a bit of a guess regarding testing, but maybe move the actual generation of the JSON object to `EBEvent` and then you can assert against those fields in `EbEventTest`?